### PR TITLE
Fixed TASK-823

### PR DESCRIPTION
**Pull request by sujata-m**
Fixed [Task-823](https://dev.azure.com/TDEI-UW/TDEI/_workitems/edit/823/)

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -11,6 +11,8 @@ class Settings(BaseSettings):
     incoming_topic_subscription: str = os.environ.get('CONFIDENCE_REQ_SUB', '')
     outgoing_topic_name: str = os.environ.get('CONFIDENCE_RES_TOPIC', '')
     storage_container_name: str = os.environ.get('CONTAINER_NAME', 'osw')
+    username: str = os.environ.get('USERNAME', 'dev@gaussiansolutions.com')
```
Comment (sureshgaussian) [inline src/config.py]: Please remove the default credentials from here. Never expose passwords in the code. If the USERNAME/PASSWORD are not provided, just exit the code saying those two are mandatory.

File: src/service/test.py
Comment (sureshgaussian) [inline src/service/test.py:1]: what is this file? why is a test under src folder? If we release this file on public git, we should not have names like DifferentClass. Fix this.

File: src/service/osw_confidence_service.py [hunk 2]
```diff
@@ -44,39 +46,47 @@ def process(self, msg: QueueMessage):
...
         # creating a dummy response now
 
         response = ConfidenceResponse(
             messageId=request.messageId,
             messageType=request.messageType,
             data=ResponseData(
                 jobId=jobId,
-                confidence_level='90.0',
+                confidence_level=score,
                 confidence_library_version='v1.0',
```
Comment (sureshgaussian) [inline src/service/osw_confidence_service.py]: this should be the version of the osw_confidence_metrics library we are using; not a static number. As the code stands today I believe this should be `python_confidence_metric.__version__`
I see the python module for the confidence metrics library is named osw_confidence_metric (from requirements.txt), however the usage of the library is python_confidence_metric. This is not good. Module name and its usuage should be same. Change python_confidence_metric to osw_confidence_metric.

File: src/config.py (hunk 1 above)
Comment (susrisha) [inline src/config.py]: Please remove default credentials. we will be giving out our credentials here.

File: src/service/osw_confidence_metric.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        # Save the convex hull as a new GeoJSON file
+        convex_hull_gdf.to_file(output_file, driver='GeoJSON')
+        return output_file
+
+    def calculate_score(self):
+        osm_data_handler = OSMDataHandler(username=self.settings.username, password=self.settings.password)
+        area_analyzer = AreaAnalyzer(osm_data_handler=osm_data_handler)
+        start_time = time.time()
+        score = area_analyzer.calculate_area_confidence_score(file_path=self.convex_file)
+        print("--- %s seconds ---" % (time.time() - start_time))
```
Comment (susrisha) [inline src/service/osw_confidence_metric.py]: please use logger here.

File: src/service/osw_confidence_metric.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import geopandas as gpd
+from src.config import Settings
+from src.service.helper import clean_up
+from python_confidence_metric.area_analyzer import AreaAnalyzer
+from python_confidence_metric.osm_data_handler import OSMDataHandler
+
+warnings.filterwarnings('ignore', category=DeprecationWarning)
+
+
+class OSWConfidenceMetric:
```
Comment (susrisha) [inline src/service/osw_confidence_metric.py]: Write a couple of lines about what this class does in sequence

File: src/service/osw_confidence_service.py [hunk 5]
```diff
@@ -44,39 +46,47 @@ def process(self, msg: QueueMessage):
...
-            os.makedirs(local_base_path)
+
         osw_file_local_path = os.path.join(local_base_path, 'osw.zip')  # Assuming zip file
         self.download_single_file(request.data.data_file, osw_file_local_path)
         meta_file_local_path = os.path.join(local_base_path, 'meta.json')  # Assuming json file
         self.download_single_file(request.data.meta_file, meta_file_local_path)
         # The meta file is at `meta_file_local_path`
         # The osw file is at `osw_file_local_path`
         # insert code for calculation here.
+        metric = OSWConfidenceMetric(zip_file=osw_file_local_path)
```
Comment (susrisha) [inline src/service/osw_confidence_service.py]: If the confidence calculation only relies on osw file, comment out the download of meta file. Its an additional performance burden to download that file.

File: src/service/test.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+from osw_confidence_metric  import OSWConfidenceMetric
+class DifferentClass:
```
Comment (susrisha) [inline src/service/test.py]: Rename to `OSWConfidenceMetricTestClass`

File: src/service/test.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    def __init__(self):
+        pass
+
+    def my_function(self):
```
Comment (susrisha) [inline src/service/test.py]: rename function to `test_osw_confidence_metric_calculation`

**Review by susrisha — CHANGES_REQUESTED** [review]
Some changes needed in documenting and renaming stuff.

**Review by susrisha — DISMISSED** [review]

**Review by sureshgaussian — CHANGES_REQUESTED** [review]

**Review by susrisha — DISMISSED** [review]

**Review by susrisha — DISMISSED** [review]

**Review by susrisha — APPROVED** [review]

**Review by sureshgaussian — APPROVED** [review]
approved after code review.

**Pull request merged**